### ISSUE TITLE
Documentation for ProcessMonitor is slightly unclear

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3961]
|<img alt="slyons's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| slyons reported|
|-|-|
|Trac ID|trac#3961|
|Type|enhancement|
|Created|2009-08-10 15:02:30Z|

As suggested by spiv in #twisted, I'm filing a ticket in regards to the documentation for ProcessMonitor. Glancing over the documentation, I assumed that one could use the startProcess and stopProcess methods to control the running process, when really they are meant to be used for internal purposes only. The docstring makes things clearer, and it would be best if some of that clarity made its way into the docs for ProcessMonitor.

<details><summary>Searchable metadata</summary>

```
trac-id__3961 3961
type__enhancement enhancement
reporter__slyons slyons
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__runner runner
keywords__documentation documentation
time__1249916550000000 1249916550000000
changetime__1264480875000000 1264480875000000
version__None None
owner__jesstess jesstess
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to spiv|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|spiv commented|
|-|-|

FWIW, the specific problems that occurred to me just glancing at that module:

 * no class docstring for ProcessMonitor;
 * no docstring on removeProcess;
 * addProcess docstring doesn't link to removeProcess;
 * most of the content in the module docstring probably belongs in the class docstring, where it would be closer to where people would expect to find it;
 * the docstrings that exist don't use the usual structured markup for describing attributes etc.

As with all documentation there's probably an infinite amount of improvement possible, but I think fixing the above would bring the docs from "barely adequate" up to "fairly good".  Most of the necessary content already exists really, it just needs to be rearranged and tweaked.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [28025]) Move comments at the top of procmon.py into the ProcessMonitor class
docstring and addProcess and removeProcess docstrings.

refs [#3961](https://github.com/twisted/twisted/issues/3961)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess removed owner|
|-|-|

Looks like the branching commit got dropped...

r28024: branched to procmon_docs_3961

spiv's comments are addressed in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @jesstess|
|-|-|

Underscores don't get noticed by the commit hook. :)

Just one comment.  The module should retain some docstring.  It doesn't need to be much.  Something like _Support for starting, monitoring, and restarting child process._ would be fine.

Fix that and then please merge.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

(In [28070]) Add a module docstring.

refs [#3961](https://github.com/twisted/twisted/issues/3961)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set status to `closed`|
|-|-|

(In [28071]) Merge procmon_docs_3961

Author: jesstess
Reviewers: exarkun
Fixes: [#3961](https://github.com/twisted/twisted/issues/3961)

Move information from the procmon module docstring into the
appropriate methods in ProcessMonitor to clarify the API.
